tests/unit/backend/wmg/api/test_v1.py

```diff
@@ -776,6 +777,65 @@ def test__query_request_with_filter_dims__returns_valid_filter_dims(
+                        organism_ontology_term_id="organism_ontology_term_id_0",
+                        tissue_ontology_term_ids=["tissue_ontology_term_id_0"],
+                        dataset_ids=["dataset_id_0"],
+                        disease_ontology_term_ids=["disease_ontology_term_id_0"],
+                        sex_ontology_term_ids=["sex_ontology_term_id_0"],
+                        development_stage_ontology_term_ids=["development_stage_ontology_term_id_0"],
+                        ethnicity_ontology_term_ids=["ethnicity_ontology_term_id_0"],
+                    )
+
+                    full_filters_request = dict(
+                        filter=full_filters,
+                        include_filter_dims=True,
+                    )
+                    self.app.post("/wmg/v1/query", json=full_filters_request)
+                    self.assertEqual(mock_dims.call_count, 5)
+
+                    mock_dims.reset_mock()
+                    no_secondary_filters = dict(
+                        gene_ontology_term_ids=["gene_ontology_term_id_0"],
+                        organism_ontology_term_id="organism_ontology_term_id_0",
+                        tissue_ontology_term_ids=["tissue_ontology_term_id_0"],
+                        dataset_ids=[],
+                        disease_ontology_term_ids=[],
+                        sex_ontology_term_ids=[],
+                        development_stage_ontology_term_ids=[],
+                        ethnicity_ontology_term_ids=[],
+                    )
+
+                    no_secondary_filters_request = dict(
+                        filter=no_secondary_filters,
+                        include_filter_dims=True,
+                    )
+                    self.app.post("/wmg/v1/query", json=no_secondary_filters_request)
+                    mock_dims.assert_not_called()
+                    # technically not necessary
+                    mock_dims.reset_mock()
+
+                    two_secondary_filters = dict(
+                        gene_ontology_term_ids=["gene_ontology_term_id_0"],
+                        organism_ontology_term_id="organism_ontology_term_id_0",
+                        tissue_ontology_term_ids=["tissue_ontology_term_id_0"],
+                        dataset_ids=[],
+                        disease_ontology_term_ids=[],
+                        sex_ontology_term_ids=[],
+                        development_stage_ontology_term_ids=["development_stage_ontology_term_id_0"],
+                        ethnicity_ontology_term_ids=["ethnicity_ontology_term_id_0"],
+                    )
+                    two_secondary_filters_request = dict(
+                        filter=two_secondary_filters,
+                        include_filter_dims=True,
+                    )
+                    self.app.post("/wmg/v1/query", json=two_secondary_filters_request)
+                    self.assertEqual(mock_dims.call_count, 2)
```

Comment:
nit: would break this up into 3 (?) separate subtests so we can see at structural level what behaviors are being tested; not 100% sure the refactoring will be cleaner tho.